src/types.ts

```diff
@@ -302,13 +310,34 @@ export type RequestMessage<T = keyof OBSRequestTypes> = T extends keyof OBSReque
 	requestData: OBSRequestTypes[T];
 } : never;
 
+export type RequestBatchRequest<T = keyof OBSRequestTypes> = T extends keyof OBSRequestTypes ? OBSRequestTypes[T] extends never ? {
+	requestType: T;
+	requestId?: string;
+} : {
+	requestType: T;
+	requestId?: string;
+	requestData: OBSRequestTypes[T];
+} : never;
```

Comment:
This feels really dirty, but without it, requests that have `never` as the `requestData` value still want a `requestData` parameter, which is an impossible constraint to satisfy. So this is the compromise.